go/fn/result.go

```diff
@@ -299,3 +299,21 @@ func ConfigObjectResult(msg string, obj *KubeObject, severity Severity) *Result
 		},
 	}
 }
+
+func ConfigObjectResultWithField(msg string, obj *KubeObject, severity Severity, field *Field) *Result {
```

Comment:
I am hoping this function can be smarter: user provides the field `path` and `proposedValue`, it can find out the `currentValue` from the `path`, determining the right type, and validate the `proposedValue` field

Also. ln304-317 can be simplified by calling `ConfigObjectResult`

Reply:
I think it is difficult to have a path and determine its `currentValue`, the path might contain a slice. How should I represent the slice using `path`? Should I add [] in the path?

Reply:
Suggest reading the [KRM function specification](https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/api-conventions/functions-spec.md) as I mentioned in last 1:1. 

**Path is the JSON path of the field e.g. `spec.template.spec.containers[3].resources.limits.cpu`**

Reply:
On second thought, this is suppose to use when logging results. All the results should be updated before the logging. There is no point getting `currentValue` using `path` then @yuwenma . In implementation, the old value and new value are stored in some map, then later used in the logging.